### PR TITLE
Add a cache strategy enum

### PR DESCRIPTION
**Pull request by Gudahtt**
An enum has been added to represent the cache strategy that we should use for each method.

### CODE REVIEW

File: src/block-cache.ts [hunk 1]
```diff
@@ -150,10 +151,11 @@ export function createBlockCacheMiddleware({
 
   // create caching strategies
   const blockCache: BlockCacheStrategy = new BlockCacheStrategy();
-  const strategies: Record<string, BlockCacheStrategy> = {
-    perma: blockCache,
-    block: blockCache,
-    fork: blockCache,
+  const strategies: Record<CacheStrategy, BlockCacheStrategy | undefined> = {
```
Comment (mcmire) [inline src/block-cache.ts:154]: Does the type need to be specified here? Is it just for explicitness?

**Review by mcmire — APPROVED** [review]
One question but looks good regardless.

File: src/block-cache.ts (hunk 1 above)
Comment (Gudahtt) [reply src/block-cache.ts:154]: It was pre-existing; I hadn't really considered whether it was needed or not, I just made the required updates to it. This library is full of unnecessary types.

Comment (Gudahtt) [reply src/block-cache.ts:154]: Looks like when I remove this, the inferred type no longer uses `CacheStrategy` as the keys. It gets reduced to the strings instead. Maybe better to preserve the enum as the key so it's clear that it includes all strategies.

**Comment by Gudahtt** [comment]
Rebased to resolve a conflict

File: src/block-cache.ts (hunk 1 above)
Comment (mcmire) [reply src/block-cache.ts:154]: Interesting. No problem.

**Review by mcmire — APPROVED** [review]
Looks good!

**Review by mcmire — APPROVED** [review]

**Pull request merged**